tuned/daemon/controller.py

```diff
@@ -417,3 +417,74 @@ def instance_get_devices(self, instance_name, caller = None):
 		rets = "Instance '%s' not found" % instance_name
 		log.error(rets)
 		return (False, rets, [])
+
+	@exports.export("ssa{ss}", "bs")
+	def instance_create(self, plugin_name, instance_name, options, caller = None):
+		"""Dynamically create a plugin instance
+
+		Parameters:
+		plugin_name -- name of the plugin
+		instance_name -- name of the new instance
+		dict of string-string -- options for the new instance
+
+		Return:
+		bool -- True on success
+		string -- error message or "OK"
+		"""
+		plugins = {p.name: p for p in self._daemon._unit_manager.plugins}
+		if not plugin_name in plugins.keys():
+			rets = "Plugin '%s' not found" % plugin_name
+			log.error(rets)
+			return (False, rets)
+		plugin = plugins[plugin_name]
+		instance = plugin.create_instance(instance_name, None, None, None, None, options)
+		plugin.initialize_instance(instance)
+		instance._created_dynamically = True
+		self._daemon._unit_manager.instances.append(instance)
+		log.info("Created dynamic instance '%s' of plugin '%s'" % (instance_name, plugin_name))
+		return (True, "OK")
+
+	@exports.export("ss", "bs")
+	def instance_destroy(self, plugin_name, instance_name, caller = None):
+		"""Destroy a dynamically created plugin instance
+
+		Parameters:
+		plugin_name -- name of the plugin
+		instance_name -- name of the new instance
+
+		Return:
+		bool -- True on success
+		string -- error message or "OK"
+		"""
+		instance = None
+		for i in self._daemon._unit_manager.instances:
+			if i.name == instance_name:
+				instance = i
+				break
+		if instance is None:
+			rets = "Instance '%s' not found" % instance_name
+			log.error(rets)
+			return (False, rets)
+		if not instance._created_dynamically:
```

Comment:
I think you have to define `_created_dynamically` in the instance constructor, otherwise it may traceback in case you try to destroy statically created instance:
```
('g-io-error-quark: GDBus.Error:org.freedesktop.DBus.Python.AttributeError: '
 "AttributeError: 'Instance' object has no attribute '_created_dynamically'\n"
 ' (36)')
```

Reply:
This could happen if destroying instance of other type than `irq`, which has the initialization.

Reply:
I removed the `_created_dynamically` flag, making dynamically created instances equal to the "static" ones defined in the profile.